scripts/setup.sh

```diff
@@ -85,11 +91,21 @@ setup_haskell() {
   # Install all solved dependency versions for the helper binaries, while skipping local dependencies package and GHC.
   # Also skipping bogus 'invalid-cabal-flag-settings' dependency from base: https://github.com/commercialhaskell/stack/issues/2969
   local HELPER_BINARIES_DEPENDENCY_LIST=$(stack list-dependencies --separator - | grep -vE "^dependencies-|^ghc-[0-9]\.[0-9]\.[0-9]$|^invalid-cabal-flag-settings-|^rts-")
-  for dep in $HELPER_BINARIES_DEPENDENCY_LIST
-  do
-    stack install $dep ; RETCODE=$?
-    [ ${RETCODE} -ne 0 ] && exit_err "Installing helper binary dependency $dep failed with error ${RETCODE}."
-  done
+
+  if [ "$DRY_RUN" == false ]
```

Comment:
Is it possible to move the conditional inside the loop to avoid duplicating the loop?